lib/beaker/hypervisor/fog.rb

```diff
@@ -0,0 +1,104 @@
+require 'fog'
+
+module Beaker
+  #Beaker support for Fog
+  class Fog < Beaker::Hypervisor
+
+    SLEEPWAIT = 5
+
+    #Create a new instance of the Fog hypervisor object
+    #@param [<Host>] hosts The array of Fog hosts to provision
+    #@param [Hash{Symbol=>String}] options The options hash containing configuration values
+    def initialize(hosts, options)
+      @options = options
+      @logger = options[:logger]
+      @hosts = hosts
+      @vms = []
+
+      host = hosts.first
+
+      fog_auth = get_cloud_auth_method(hosts)
+
+      fog_compute_options = { :provider => host[:cloud_provider]}.merge( fog_auth )
+
+      @compute_client ||= ::Fog::Compute.new( fog_compute_options )
+
+      if not @compute_client
+        raise "Unable to create #{host[:cloud_provider]} Compute instance"
+      end
+    end
+
+    #Create new instances in digitalocean
+    def provision
+      @hosts.each do |host|
+        @logger.notify "Provisioning #{host[:cloud_provider]} using Fog"
+        host[:vmhostname] = generate_host_name
+        @logger.debug "Provisioning #{host.name} (#{host[:vmhostname]})"
+
+        extras = provider_specific_fields(host)
+
+        default_server_options = {
+          :name => host[:vmhostname],
+          :image_id => host[:image_id],
+          :public_key_path => host[:public_key_path] || File.expand_path('~/.ssh/id_rsa.pub'),
```

Comment:
Default key values should be stored in options/presets.rb for easier tracking across the project.